CMakeLists.txt

```diff
@@ -0,0 +1,345 @@
+set(GCC_MD5 "2d19775c5aa091b7f55a68727c5d7d76" CACHE STRING "GCC MD5 Hash")
+set(GCC_FOLDER ${CMAKE_SOURCE_DIR}/external/gcc CACHE PATH "GCC Path")
+file(MAKE_DIRECTORY ${GCC_FOLDER})
+
+set(TONC_URL "http://www.coranac.com/files/tonc-code.zip")
+set(TONC_MD5 "3f9ebdbbe7989b33c2a788c5b5a1bfed" CACHE STRING "TONC MD5 Hash")
+set(TONC_FOLDER ${CMAKE_SOURCE_DIR}/external/tonc CACHE PATH "TONC Path")
+file(MAKE_DIRECTORY ${TONC_FOLDER})
+
+set(LSRC_VERSION "0.1.9" CACHE STRING "LibSampleRate Version")
+set(LSRC_URL "http://www.mega-nerd.com/SRC/libsamplerate-${LSRC_VERSION}.tar.gz")
+set(LSRC_MD5 "2b78ae9fe63b36b9fbb6267fad93f259" CACHE STRING "LibSampleRate MD5 Hash")
+set(LSRC_INSTALL_FOLDER ${CMAKE_SOURCE_DIR}/external/libsamplerate-install CACHE PATH "LibSampleRate Install Path")
+set(LSRC_SOURCE_FOLDER ${CMAKE_SOURCE_DIR}/external/libsamplerate CACHE PATH "LibSampleRate Source Path")
+file(MAKE_DIRECTORY ${LSRC_SOURCE_FOLDER})
+
+# Resource File Definitions
+file(GLOB_RECURSE SPR_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/data/sprites/ ${PROJECT_SOURCE_DIR}/data/sprites/*.png)
+file(GLOB_RECURSE BKG_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/data/backgrounds/ ${PROJECT_SOURCE_DIR}/data/backgrounds/*.png)
+file(GLOB_RECURSE FNT_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/data/fonts/ ${PROJECT_SOURCE_DIR}/data/fonts/*.ttf)
+set(AUS_FILES
+	audio.json) # Not using GLOB here since its just a single file, and for audio. There seems to be some other json files in the data directories too, but they are used for bitmask stuff?
+file(GLOB_RECURSE SND_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/data/sounds/ ${PROJECT_SOURCE_DIR}/data/sounds/*.wav)
+
+# Source files
+file(GLOB_RECURSE IWRAM_C_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/source ${PROJECT_SOURCE_DIR}/source/*.iwram.c)
+file(GLOB_RECURSE IWRAM_CPP_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/source ${PROJECT_SOURCE_DIR}/source/*.iwram.cpp)
+file(GLOB_RECURSE GAME_C_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/source ${PROJECT_SOURCE_DIR}/source/*.c)
+list(FILTER GAME_C_FILES EXCLUDE REGEX ".*iwram.c$")
+file(GLOB_RECURSE GAME_CPP_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/source ${PROJECT_SOURCE_DIR}/source/*.cpp)
+list(FILTER GAME_CPP_FILES EXCLUDE REGEX ".*iwram.cpp$")
+file(GLOB_RECURSE GAME_ASM_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/source ${PROJECT_SOURCE_DIR}/source/*.s)
+
+# Build Functions
+function(build_resource)
+	# Set Variables to make this easier
+	# i.e SPRITE
+	set(RESOURCE_NAME ${ARGV0})
+	# i.e SPRITES
+	set(PLURAL ${RESOURCE_NAME}S)
+	# passed in dependencies, tool or the previous object
+	set(TARGET_DEPENDENCIES ${ARGV1})
+	# Set helped variables based on what resource we are exporting and building
+	if(${RESOURCE_NAME} MATCHES "SPRITE")
```

Comment:
Instead of setting `OUT1` and `OUT2`, you can also export a single list variable `OUT` with the output extensions generated for each tool. Helps if some tools are generated in the future that export more than two outputs.

Reply:
Yeah, that makes sense.